client/src/app/site/pages/organization/pages/accounts/pages/account-detail/components/account-add-to-meetings/account-add-to-meetings.component.ts

```diff
@@ -0,0 +1,164 @@
+        return this.assignMeetingsForm.get(`meeting_ids`).value as number[];
+    }
+
+    public get groupName(): string {
+        return this.assignMeetingsForm.get(`group_name`).value as string;
+    }
+
+    public get hasResult(): boolean {
+        let dataLength = 0;
+        Object.keys(this.resultsTableDataSubjects).forEach(key => {
+            dataLength = dataLength + this.resultsTableDataSubjects[key].value.length;
+        });
+        return !!dataLength;
+    }
+
+    public resultsTableDataSubjects = {
+        succeeded: new BehaviorSubject<string[]>([]),
+        standard_group: new BehaviorSubject<string[]>([]),
+        nothing: new BehaviorSubject<string[]>([])
+    };
+
+    public get waitingForResults(): boolean {
+        return this._waitingForResults;
+    }
+
+    private _waitingForResults = false;
+
+    private userId: Id | null = null;
+
+    public constructor(
+        private operator: OperatorService,
+        private userController: UserControllerService,
+        private meetingController: MeetingControllerService,
+        private router: Router,
+        private osRouter: OpenSlidesRouterService,
+        private formBuilder: UntypedFormBuilder
+    ) {
+        super();
+    }
+
+    public ngOnInit(): void {
+        this.subscriptions.push(
+            this.osRouter.currentParamMap.subscribe(params => {
+                if (params[`id`]) {
+                    this.userId = +params[`id`];
+                    this.loadUser();
+                }
+            }),
+            this.meetingController
+                .getViewModelListObservable()
+                .subscribe(meetings => this.meetingsSubject.next(meetings))
+        );
+    }
+
+    public ngAfterViewInit(): void {
+        setTimeout(() => this.updatePermissions()); // Initial check
+    }
+
+    public async assign(): Promise<void> {
+        if (this.user) {
+            this._waitingForResults = true;
+            const result = await this.userController
+                .assignMeetings(this.user, { meeting_ids: this.selectedMeetings, group_name: this.groupName })
+                .resolve();
+            if (result) {
+                this.parseIntoResultSubject(result);
+                this.clearForm();
+            }
+            this._waitingForResults = false;
+        }
+    }
+
+    private clearForm(): void {
+        this.assignMeetingsForm.get(`group_name`).reset();
+        this.assignMeetingsForm.get(`meeting_ids`).reset([]);
+    }
+
+    /**
+     * Triggered by the "x" Button of the Form
+     */
+    public goBack(): void {
+        if (this.user) {
+            this.router.navigate([`accounts`, this.user.id]);
+        }
+    }
+
+    private parseIntoResultSubject(data: AssignMeetingsResult[]): void {
+        Object.keys(this.resultsTableDataSubjects).forEach(key => {
+            this.resultsTableDataSubjects[key].next([]);
+        });
+        data.forEach(result => {
```

Comment:
You can be sure that there is always exactly one result, so you don't need to iterate over the data or do the whole concatenation below, if I understood it right. Just directly mapping ids to names and feeding it into the subject should be enough.